ui/src/heap/HeapChannel.tsx

```diff
@@ -197,22 +248,18 @@ function HeapChannel({ title }: ViewProps) {
             : title}
         </title>
       </Helmet>
-      <div className="h-full overflow-y-scroll p-4">
-        <div
-          className={cn(
+      <div className="h-full p-4">
+        <VirtuosoGrid
+          data={canWrite ? sortedCurios : sortedCurios.slice(1)}
+          itemContent={(i, [time, curio]) => renderCurio(i, curio, time)}
+          computeItemKey={computeItemKey}
+          style={{ height: '100%', width: '100%', paddingTop: '1rem' }}
+          atBottomStateChange={loadOlderCurios}
```

Comment:
this fires on both entry and leaving bottom

Reply:
Good catch, fixed in latest commit.